src/main/java/org/broadinstitute/hellbender/engine/dataflow/datasources/ReadsDataflowSource.java

```diff
@@ -143,6 +149,25 @@ public SAMFileHeader getHeader() {
     }
 
     public static PCollectionView<SAMFileHeader> getHeaderView(Pipeline p, SAMFileHeader readsHeader) {
-        return p.apply(Create.of(readsHeader).withCoder(SerializableCoder.of(SAMFileHeader.class))).apply(View.<SAMFileHeader>asSingleton());
+        return p.apply(Create.of(readsHeader).withCoder(SerializableCoder.of(SAMFileHeader.class))).setName("reads header").apply(View.<SAMFileHeader>asSingleton());
     }
+
+
+
+    /**
+     * Create a {@link PCollection< KV <String,Iterable<GATKRead>>>} containing all the reads starting in the given intervals and their mates,
+     * grouped by the shard they start in.
+     * The shards are strings as per SimpleInterval's ToString syntax.
+     * Reads that are malformed or unmapped are ignored.
+     *
+     * Note: for performance reasons, this will give you SAMRecords that have no header. You have to put it back in.
```

Comment:
This warning needs to be way more prominent -- the reads will be returned in an **unusable** state for many operations until the header is restored.